core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -87,13 +89,23 @@ class SnapshotService(protected val userInfo: UserInfo, val dataSource: SlickDat
     }
   }
 
-  def enumerateSnapshots(workspaceName: WorkspaceName, offset: Int, limit: Int): Future[ResourceList] = {
+  def massageSnapshots(references: ResourceList): Seq[DataRepoSnapshotResource] = {
+    val res = new ListBuffer[DataRepoSnapshotResource]()
+    references.getResources.forEach(r => { val massaged = new DataRepoSnapshotResource
```

Comment:
I want to do a references.getResources.map one liner, but it seems like there's some java/scala interop fu that I'm missing as there's no map method on `getResources`

Reply:
`getResources` returns a `java.util.List`, so Scala's `map` isn't available. The java/scala interop fu you're looking for is `import scala.collection.JavaConverters._` … see https://docs.scala-lang.org/overviews/collections/conversions-between-java-and-scala-collections.html for details. Once you've imported that, you have `asScala` available to you, and you can do:
```
    references.getResources.asScala.map{r =>
      val massaged = new DataRepoSnapshotResource
      massaged.setAttributes(r.getResourceAttributes.getGcpDataRepoSnapshot)
      massaged.setMetadata(r.getMetadata)
      massaged
    }
```

this eliminates the use of the `mutable.ListBuffer`, which is good - any mutable classes are un-idiomatic in Scala (of course *_sometimes_* you need one, but it's the exception)